src/AutoMapper/TypeMap.cs

```diff
@@ -110,6 +110,7 @@ public PathMap FindOrCreatePathMapFor(LambdaExpression destinationExpression, Me
             || CustomCtorFunction != null
             || ConstructDestinationUsingServiceLocator
             || ConstructorMap?.CanResolve == true
+            || DestinationTypeToUse.IsArray
```

Comment:
This seems very specific. Why doesn't it fail for the no-arg ctor check further down?

Reply:
> This used to work because ConstructorMap.CanResolve was broken. An array doesn't have a default constructor.

I don't have a better explanation and I'm not sure what you mean :)

Reply:
What is special about arrays?

Reply:
They don't have a default constructor, but there is a special function to create them. And we do use that at runtime, just the validation fails.

Reply:
Actually I'm not sure we would use that for a custom map. The test works because a destination is passed in. So just fail?